src/DIRAC/Core/Security/test/Test_X509Chain.py

```diff
@@ -529,51 +516,50 @@ def test_delegation(get_X509Request, get_proxy, diracGroup, lifetime):
 
   # The server side generates a request
   # Equivalent to ProxyManager.requestDelegationUpload
-  x509Req = get_X509Request()
-  x509Req.generateProxyRequest()
-  reqStr = x509Req.dumpRequest()['Value']
-
-  # This object contains both the public and private key
-  pkeyReq = x509Req.getPKey()
+  with get_X509Request(x509ClassName) as x509Req:
+    x509Req.generateProxyRequest()
+    reqStr = x509Req.dumpRequest()['Value']
 
-  #######################################################
+    # This object contains both the public and private key
+    pkeyReq = x509Req.getPKey()
 
-  # The client side signs the request
+    #######################################################
 
-  proxyChain = get_proxy(USERCERT, diracGroup=diracGroup)
+    # The client side signs the request
 
-  # The proxy will contain a "bullshit private key"
-  res = proxyChain.generateChainFromRequestString(reqStr, lifetime=lifetime)
+    with get_proxy(x509ChainClassName, USERCERT, diracGroup=diracGroup, deimport=False) as proxyChain:
```

Comment:
Maybe add a comment here to explain why `deimport` is False (copy paste what was in the previous incarnation of `get_proxy` ? )